### PR TITLE
[Services Purify 2] Initial Purify effort

### PR DESCRIPTION
**Pull request by MatApple**
This PR is huge, I know. Since the goal is to merge this on the plane to Hamburg it's going to be easier as a single PR.

This PR:
* Refactors the ServicesTab to now be ServicesContainer which is control of fetching data and handling actions.
* Refactors the filters so they are a little more robust
* Refactors services modals so they are presentational components
* Moves some files into a more logical arrangement based on a single container per directory

*Note*
* Somehow Git lost some of the git mv commands and I've commented on those files so you know their is a counterpart "add" to the "deletion".
* Service Detail and other nested routes will just show "Could not find Services" page. Those updates will come in another PR.

### CODE REVIEW

File: plugins/services/src/js/services/ServicesContainer.js [hunk 1]
```diff
@@ -0,0 +1,689 @@
...
+  constructor() {
+    super(...arguments);
+
+    this.state = {
+      actionErrors: {},
+      fetchErrors: {},
+      filters: {},
+      isLoading: true,
+      lastUpdate: 0,
+      openModalId: null,
```
Comment (rcorral) [inline plugins/services/src/js/services/ServicesContainer.js]: This could be removed.

File: plugins/services/src/js/services/SidebarFilter.js [hunk 2]
```diff
@@ -190,7 +173,7 @@ class SidebarFilter extends mixin(QueryParamsMixin) {
         formGroupClass="form-group flush"
         formRowClass="row"
         definition={definition}
-        onChange={this.handleFormChange.bind(this)} />
+        onChange={(model, e) => { this.handleFormChange(model, e); }} />
```
Comment (rcorral) [inline plugins/services/src/js/services/SidebarFilter.js]: Don't think you need to do this.

File: plugins/services/src/js/services/SidebarFilter.js [hunk 3]
```diff
@@ -157,7 +138,9 @@ class SidebarFilter extends mixin(QueryParamsMixin) {
       })
       .map((filterLabel) => {
         let value = filterValues[filterLabel];
-        let checked = selectedNodes.indexOf(value.toString()) > -1;
+        let checked = selectedNodes.indexOf(value.toString()) > -1
+          || selectedNodes.indexOf(value) > -1;
```
Comment (rcorral) [inline plugins/services/src/js/services/SidebarFilter.js]: Do we still need this?

**Comment by rcorral** [comment]
@MatApple would love to see tests written for some of the new code that makes sense. 

Also looks like tests are failing, though haven't ran them locally.

File: plugins/services/src/js/components/__tests__/ServicesTable-test.js [hunk 4]
```diff
@@ -24,10 +24,17 @@ describe('ServicesTable', function () {
...
+    restartService() {},
+    suspendService() {},
+    deleteService() {}
+  };
+
   beforeEach(function () {
     this.container = document.createElement('div');
     this.instance = ReactDOM.render(
-      <ServicesTable />,
+      <ServicesTable modals={modals} />,
```
Comment (MatApple) [inline plugins/services/src/js/components/__tests__/ServicesTable-test.js]: ✔️

File: npm-shrinkwrap.json [hunk 5]
```diff
@@ -2825,6 +2825,613 @@
...
+          "from": "wrappy@>=1.0.0 <2.0.0",
+          "resolved": "https://registry.npmjs.org/wrappy/-/wrappy-1.0.2.tgz"
+        },
+        "xtend": {
+          "version": "4.0.1",
+          "from": "xtend@>=4.0.0 <5.0.0",
+          "resolved": "https://registry.npmjs.org/xtend/-/xtend-4.0.1.tgz"
+        }
+      }
+    },
```
Comment (MatApple) [inline npm-shrinkwrap.json]: ✔️

File: plugins/services/src/js/components/modals/KillPodInstanceModal.js [hunk 6]
```diff
@@ -124,42 +74,61 @@ class KillPodInstanceModal extends mixin(StoreMixin) {
...
-  action: React.PropTypes.string,
-  onClose: React.PropTypes.func.isRequired,
-  onSuccess: React.PropTypes.func,
-  open: React.PropTypes.bool.isRequired,
-  pod: React.PropTypes.instanceOf(Pod),
-  selectedItems: React.PropTypes.array.isRequired
+  action: PropTypes.string,
+  errors: PropTypes.string,
+  killPodInstances: PropTypes.func.isRequired,
+  isPending: PropTypes.bool.isRequired,
```
Comment (MatApple) [inline plugins/services/src/js/components/modals/KillPodInstanceModal.js]: ✔️

File: plugins/services/src/js/components/modals/KillPodInstanceModal.js [hunk 7]
```diff
@@ -124,42 +74,61 @@ class KillPodInstanceModal extends mixin(StoreMixin) {
...
+  onClose: PropTypes.func.isRequired,
+  open: PropTypes.bool.isRequired,
+  pod: PropTypes.instanceOf(Pod),
```
Comment (MatApple) [inline plugins/services/src/js/components/modals/KillPodInstanceModal.js]: ✔️

File: plugins/services/src/js/components/modals/KillTaskModal.js [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+    if (!errors) {
+      return null;
+    }
+
+    if (this.shouldForceUpdate()) {
+      return (
+        <h4 className="text-align-center text-danger flush-top">
+          App is currently locked by one or more deployments. Press the button
+          again to forcefully change and deploy the new configuration.
+        </h4>
```
Comment (MatApple) [inline plugins/services/src/js/components/modals/KillTaskModal.js]: ✔️

File: plugins/services/src/js/components/modals/KillTaskModal.js [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+KillTaskModal.defaultProps = {
+  action: 'kill',
+  selectedItems: []
+};
+
+KillTaskModal.propTypes = {
+  action: PropTypes.string,
+  errors: PropTypes.string,
+  killTasks: PropTypes.func.isRequired,
+  isPending: PropTypes.bool.isRequired,
```
Comment (MatApple) [inline plugins/services/src/js/components/modals/KillTaskModal.js]: ✔️

File: plugins/services/src/js/components/modals/ServiceModals.js [hunk 10]
```diff
@@ -0,0 +1,334 @@
...
+  editGroup: PropTypes.func,
+  createService: PropTypes.func,
+  deleteService: PropTypes.func,
+  editService: PropTypes.func,
+  restartService: PropTypes.func,
+  killTasks: PropTypes.func
+}).isRequired;
+
+ServiceModals.propTypes = {
+  actionErrors: PropTypes.object,
```
Comment (MatApple) [inline plugins/services/src/js/components/modals/ServiceModals.js]: ✔️

File: plugins/services/src/js/components/modals/ServiceRestartModal.js [hunk 11]
```diff
@@ -63,4 +84,16 @@ class ServiceRestartModal extends ServiceActionModal {
   }
 }
 
+ServiceRestartModal.propTypes = {
+  restartService: PropTypes.func.isRequired,
```
Comment (MatApple) [inline plugins/services/src/js/components/modals/ServiceRestartModal.js]: ✔️

File: plugins/services/src/js/services/SidebarFilter.js (hunk 2 above)
Comment (MatApple) [inline plugins/services/src/js/services/SidebarFilter.js]: ✔️

File: plugins/services/src/js/services/SidebarFilter.js (hunk 3 above)
Comment (MatApple) [inline plugins/services/src/js/services/SidebarFilter.js]: Actually, we do 👍 because we could get filter values passed down as ints or strings.

**Comment by MatApple** [comment]
It's not actually the tests that failed - unexpected eof after npm i.

File: plugins/services/src/js/components/modals/AppLockedMessage.js [hunk 12]
```diff
@@ -0,0 +1,14 @@
+/* eslint-disable no-unused-vars */
+import React from 'react';
+/* eslint-enable no-unused-vars */
+
+const AppLockedMessage = function () {
+  return (
+    <h4 className="text-align-center text-danger flush-top">
+      App is currently locked by one or more deployments. Press the button
```
Comment (MatApple) [inline plugins/services/src/js/components/modals/AppLockedMessage.js]: Yip, just copied what existed. I'll update it to 'Service'

Comment (MatApple) [inline plugins/services/src/js/components/modals/AppLockedMessage.js]: ✔️

**Comment by MatApple** [comment]
Addressed all comments and wrote some more tests 👍

**Comment by rcorral** [comment]
Can we just delete that test file? So that we don't have a bunch of commented out code?

**Pull request merged**